metrics/metrics-tests/src/main/java/org/apache/james/metrics/tests/RecordingMetricFactory.java

```diff
@@ -65,6 +67,13 @@ public TimeMetric timer(String name) {
         return Flux.from(publisher).doOnComplete(timer::stopAndPublish);
     }
 
+    @Override
+    public <T> Publisher<T> runPublishingTimerMetricLogP99(String name, Publisher<T> publisher) {
+        TimeMetric timer = timer(name);
```

Comment:
Can't this be done in a default method?

Reply:
Unfortunately not really, as we excluded Reactor from Apis, and `MetricFactory` is one